### DISCUSSION TITLE
Nested serializer source argument fails silently: field missing

### DISCUSSION THREAD

**Discussion by dennisvang** [encode/django-rest-framework#9536] [Potential Issue]
This looks like a bug, but maybe there's some documentation that I missed.

Suppose I have a model serializer with a field based on a nested serializer:

```python
class MyModelSerializer(serializers.ModelSerializer):
    class Meta:
        model = MyModel
        fields = ('my_field')

    my_field = BarSerializer(
        source='foo.bar_set', many=True, read_only=True
    )
```

Normally this works.

However, there are (afaik) two cases in which `my_field` is *completely omitted* from the serialized data:

- if there is a typo in the `source` argument (e.g. I erroneously tried to use django lookup syntax: `'foo__bar_set'`)
- if there is no related `foo` object,  i.e. if `MyModel.foo` is `None`

There is ***no error*** to warn about the missing field.

Using DRF 3.15.2 on windows 10 or linux, python 3.11.

Also see [here][1].

[1]: https://stackoverflow.com/a/78943976

**Comment by lovelydinosaur** [comment]
https://github.com/encode/django-rest-framework/blob/master/rest_framework/fields.py#L431

I had to remind myself from the source what's going on here. Looks like you'll get an error if the field is required otherwise it'll be omitted. I'd suggest this neither is nor isn't a bug, it's just fuzzy behavior.

**Reply by dennisvang** [reply]
@tomchristie Thanks for looking into this!

For me it was quite surprising that the entire field is omitted, instead of e.g. including the field with a value of `None`.

This caused a bug in my code, which, imho, would have been much easier to track down if there had been a warning or error message.

Anyway, thank you for creating this wonderful thing called DRF!
